### ISSUE TITLE
Can not get any usefull output, help appreciated (sample data provided)

### ISSUE THREAD

**Issue by devilmastah** [VROOM-Project/vroom#1029]
Edit:
I figured out that if i change the time_window:[val,val] to time_windows:[[val:val]], it suddenly makes a lot more sense, i think i misread that in the docs. 
But the output still does not make a lot of sense, tomorrow is another day have been at it for 6 hours

I am trying to solve a relative complex routing issue and can not find out if i am doing something wrong in providing my data or if i misunderstand the whole logic. 

Example data is under this message. 

I have a fleet of 8 vehicles all basicly have a service time that is the whole day, defined as seconds from midnight. 

And i have a range of orders that have:
- a window of time in wich the order can start, this is a 30 minute window, both defined as seconds from midnight
- a duration in seconds

When i run this i get a response in which only 3 cars are used for all orders and a lot of the orders overlap. 
I get no violations, no waiting time, but at least the duration and service responses seem correct. 
If anyone can point out if i am doing something wrong then i would love to hear it. 

What i tried:

- using times based unix timestamps
- Using service_after in the jobs (does not make any difference)

Other noteworthy things:
- This should be solvable, timefold solves it with even more constraints and gives 3 violations. 
- I also dont see any waiting time increasing, i dont understand that either

Input:
`{"vehicles":[{"id":0,"name":"18","profile":"driving-car","time_window":[0,86400],"start":[4.940560167,52.3031305]},{"id":1,"name":"17","profile":"driving-car","time_window":[0,86400],"start":[4.940293333,52.3040685]},{"id":2,"name":"16","profile":"driving-car","time_window":[0,86400],"start":[4.9406835,52.303656167]},{"id":3,"name":"15","profile":"driving-car","time_window":[0,86400],"start":[4.942739,52.305608]},{"id":4,"name":"14","profile":"driving-car","time_window":[0,86400],"start":[4.940863167,52.3040605]},{"id":5,"name":"13","profile":"driving-car","time_window":[0,86400],"start":[4.940999,52.303619]},{"id":6,"name":"11","profile":"driving-car","time_window":[0,86400],"start":[4.9406955,52.304431333]},{"id":7,"name":"10","profile":"driving-car","time_window":[0,86400],"start":[4.941004667,52.303224667]}],"jobs":[{"id":0,"name":"17815","location":[5.0000252,52.3506167],"service":1500,"time_window":[57600,59400],"service_after":57600},{"id":1,"name":"17935","location":[4.9606438,52.3707755],"service":7020,"time_window":[59400,61200],"service_after":59400},{"id":2,"name":"18216","location":[4.967969,52.2659098],"service":4140,"time_window":[59400,61200],"service_after":59400},{"id":3,"name":"18237","location":[4.8622871,52.3203643],"service":7020,"time_window":[59400,61200],"service_after":59400},{"id":4,"name":"18111","location":[4.8632049,52.3271261],"service":1620,"time_window":[59400,61200],"service_after":59400},{"id":5,"name":"17948","location":[4.8596939,52.3167195],"service":1140,"time_window":[59700,61500],"service_after":59700},{"id":6,"name":"18010","location":[4.9020046,52.3521347],"service":5340,"time_window":[61200,63000],"service_after":61200},{"id":7,"name":"17735","location":[4.8558705,52.3098863],"service":3180,"time_window":[61860,63660],"service_after":61860},{"id":8,"name":"18209","location":[4.8616944,52.3101017],"service":1140,"time_window":[62160,63960],"service_after":62160},{"id":9,"name":"18179","location":[4.8977438,52.298962],"service":1620,"time_window":[62880,64680],"service_after":62880},{"id":10,"name":"13433","location":[4.7437581,52.2296608],"service":4380,"time_window":[63000,64800],"service_after":63000},{"id":11,"name":"18079","location":[4.8354864,52.2930801],"service":1620,"time_window":[64380,66180],"service_after":64380},{"id":12,"name":"18113","location":[5.052125,52.3347587],"service":4500,"time_window":[64800,66600],"service_after":64800},{"id":13,"name":"18154","location":[4.8303962,52.2723105],"service":2820,"time_window":[64800,66600],"service_after":64800},{"id":14,"name":"18136","location":[5.004621,52.2867276],"service":2580,"time_window":[66600,68400],"service_after":66600},{"id":15,"name":"17799","location":[4.9014077,52.2929923],"service":1500,"time_window":[67860,69660],"service_after":67860},{"id":16,"name":"18121","location":[4.9275044,52.3536108],"service":2460,"time_window":[68400,70200],"service_after":68400},{"id":17,"name":"18221","location":[4.864766,52.2766861],"service":1020,"time_window":[68760,70560],"service_after":68760},{"id":18,"name":"17787","location":[4.8670086,52.320876],"service":1620,"time_window":[69240,71040],"service_after":69240}]}`

Output:
`
{"code":0,"summary":{"cost":8951,"routes":3,"unassigned":0,"setup":0,"service":56220,"duration":8951,"waiting_time":0,"priority":0,"violations":[],"computing_times":{"loading":76,"solving":8}},"unassigned":[],"routes":[{"vehicle":0,"cost":1167,"setup":0,"service":6720,"duration":1167,"waiting_time":0,"priority":0,"steps":[{"type":"start","location":[4.940560167,52.3031305],"setup":0,"service":0,"waiting_time":0,"arrival":0,"duration":0,"violations":[]},{"type":"job","location":[4.967969,52.2659098],"id":2,"setup":0,"service":4140,"waiting_time":0,"job":2,"arrival":558,"duration":558,"violations":[]},{"type":"job","location":[5.004621,52.2867276],"id":14,"setup":0,"service":2580,"waiting_time":0,"job":14,"arrival":5307,"duration":1167,"violations":[]},{"type":"end","location":[5.004621,52.2867276],"setup":0,"service":0,"waiting_time":0,"arrival":7887,"duration":1167,"violations":[]}],"violations":[]},{"vehicle":5,"cost":2971,"setup":0,"service":20820,"duration":2971,"waiting_time":0,"priority":0,"steps":[{"type":"start","location":[4.940999,52.303619],"setup":0,"service":0,"waiting_time":0,"arrival":0,"duration":0,"violations":[]},{"type":"job","location":[4.9020046,52.3521347],"id":6,"setup":0,"service":5340,"waiting_time":0,"job":6,"arrival":669,"duration":669,"violations":[]},{"type":"job","location":[4.9275044,52.3536108],"id":16,"setup":0,"service":2460,"waiting_time":0,"job":16,"arrival":6456,"duration":1116,"violations":[]},{"type":"job","location":[4.9606438,52.3707755],"id":1,"setup":0,"service":7020,"waiting_time":0,"job":1,"arrival":9413,"duration":1613,"violations":[]},{"type":"job","location":[5.0000252,52.3506167],"id":0,"setup":0,"service":1500,"waiting_time":0,"job":0,"arrival":16922,"duration":2102,"violations":[]},{"type":"job","location":[5.052125,52.3347587],"id":12,"setup":0,"service":4500,"waiting_time":0,"job":12,"arrival":19291,"duration":2971,"violations":[]},{"type":"end","location":[5.052125,52.3347587],"setup":0,"service":0,"waiting_time":0,"arrival":23791,"duration":2971,"violations":[]}],"violations":[]},{"vehicle":7,"cost":4813,"setup":0,"service":28680,"duration":4813,"waiting_time":0,"priority":0,"steps":[{"type":"start","location":[4.941004667,52.303224667],"setup":0,"service":0,"waiting_time":0,"arrival":0,"duration":0,"violations":[]},{"type":"job","location":[4.9014077,52.2929923],"id":15,"setup":0,"service":1500,"waiting_time":0,"job":15,"arrival":642,"duration":642,"violations":[]},{"type":"job","location":[4.8977438,52.298962],"id":9,"setup":0,"service":1620,"waiting_time":0,"job":9,"arrival":2542,"duration":1042,"violations":[]},{"type":"job","location":[4.8616944,52.3101017],"id":8,"setup":0,"service":1140,"waiting_time":0,"job":8,"arrival":4587,"duration":1467,"violations":[]},{"type":"job","location":[4.8670086,52.320876],"id":18,"setup":0,"service":1620,"waiting_time":0,"job":18,"arrival":5938,"duration":1678,"violations":[]},{"type":"job","location":[4.8622871,52.3203643],"id":3,"setup":0,"service":7020,"waiting_time":0,"job":3,"arrival":7676,"duration":1796,"violations":[]},{"type":"job","location":[4.8632049,52.3271261],"id":4,"setup":0,"service":1620,"waiting_time":0,"job":4,"arrival":14931,"duration":2031,"violations":[]},{"type":"job","location":[4.8596939,52.3167195],"id":5,"setup":0,"service":1140,"waiting_time":0,"job":5,"arrival":16737,"duration":2217,"violations":[]},{"type":"job","location":[4.8558705,52.3098863],"id":7,"setup":0,"service":3180,"waiting_time":0,"job":7,"arrival":18010,"duration":2350,"violations":[]},{"type":"job","location":[4.8354864,52.2930801],"id":11,"setup":0,"service":1620,"waiting_time":0,"job":11,"arrival":21595,"duration":2755,"violations":[]},{"type":"job","location":[4.864766,52.2766861],"id":17,"setup":0,"service":1020,"waiting_time":0,"job":17,"arrival":23712,"duration":3252,"violations":[]},{"type":"job","location":[4.8303962,52.2723105],"id":13,"setup":0,"service":2820,"waiting_time":0,"job":13,"arrival":25196,"duration":3716,"violations":[]},{"type":"job","location":[4.7437581,52.2296608],"id":10,"setup":0,"service":4380,"waiting_time":0,"job":10,"arrival":29113,"duration":4813,"violations":[]},{"type":"end","location":[4.7437581,52.2296608],"setup":0,"service":0,"waiting_time":0,"arrival":33493,"duration":4813,"violations":[]}],"violations":[]}]}`

Result: 
`Vehicle :18
18216
Start driving to order at (substracted travel time from order start time): 17:50:42
Start working on order (as taken from input): 18:00:00
Finished working on order (based on 'service and start'): 19:09:00

18136
Start driving to order at (substracted travel time from order start time): 19:40:33
Start working on order (as taken from input): 20:00:00
Finished working on order (based on 'service and start'): 20:43:00

17815
Start driving to order at (substracted travel time from order start time): 17:10:33
Start working on order (as taken from input): 17:30:00
Finished working on order (based on 'service and start'): 17:30:00

Vehicle :13
18010
Start driving to order at (substracted travel time from order start time): 18:18:51
Start working on order (as taken from input): 18:30:00
Finished working on order (based on 'service and start'): 19:59:00

18121
Start driving to order at (substracted travel time from order start time): 20:11:24
Start working on order (as taken from input): 20:30:00
Finished working on order (based on 'service and start'): 21:11:00

17935
Start driving to order at (substracted travel time from order start time): 17:33:07
Start working on order (as taken from input): 18:00:00
Finished working on order (based on 'service and start'): 19:57:00

17815
Start driving to order at (substracted travel time from order start time): 16:54:58
Start working on order (as taken from input): 17:30:00
Finished working on order (based on 'service and start'): 17:55:00

18113
Start driving to order at (substracted travel time from order start time): 18:40:29
Start working on order (as taken from input): 19:30:00
Finished working on order (based on 'service and start'): 20:45:00

17815
Start driving to order at (substracted travel time from order start time): 16:40:29
Start working on order (as taken from input): 17:30:00
Finished working on order (based on 'service and start'): 17:30:00

Vehicle :10
17799
Start driving to order at (substracted travel time from order start time): 20:10:18
Start working on order (as taken from input): 20:21:00
Finished working on order (based on 'service and start'): 20:46:00

18179
Start driving to order at (substracted travel time from order start time): 18:40:38
Start working on order (as taken from input): 18:58:00
Finished working on order (based on 'service and start'): 19:25:00

18209
Start driving to order at (substracted travel time from order start time): 18:21:33
Start working on order (as taken from input): 18:46:00
Finished working on order (based on 'service and start'): 19:05:00

17787
Start driving to order at (substracted travel time from order start time): 20:16:02
Start working on order (as taken from input): 20:44:00
Finished working on order (based on 'service and start'): 21:11:00

18237
Start driving to order at (substracted travel time from order start time): 17:30:04
Start working on order (as taken from input): 18:00:00
Finished working on order (based on 'service and start'): 19:57:00

18111
Start driving to order at (substracted travel time from order start time): 17:26:09
Start working on order (as taken from input): 18:00:00
Finished working on order (based on 'service and start'): 18:27:00

17948
Start driving to order at (substracted travel time from order start time): 17:28:03
Start working on order (as taken from input): 18:05:00
Finished working on order (based on 'service and start'): 18:24:00

17735
Start driving to order at (substracted travel time from order start time): 18:01:50
Start working on order (as taken from input): 18:41:00
Finished working on order (based on 'service and start'): 19:34:00

18079
Start driving to order at (substracted travel time from order start time): 18:37:05
Start working on order (as taken from input): 19:23:00
Finished working on order (based on 'service and start'): 19:50:00

18221
Start driving to order at (substracted travel time from order start time): 19:41:48
Start working on order (as taken from input): 20:36:00
Finished working on order (based on 'service and start'): 20:53:00

18154
Start driving to order at (substracted travel time from order start time): 18:28:04
Start working on order (as taken from input): 19:30:00
Finished working on order (based on 'service and start'): 20:17:00

13433
Start driving to order at (substracted travel time from order start time): 17:39:47
Start working on order (as taken from input): 19:00:00
Finished working on order (based on 'service and start'): 20:13:00

17815
Start driving to order at (substracted travel time from order start time): 16:09:47
Start working on order (as taken from input): 17:30:00
Finished working on order (based on 'service and start'): 17:30:00`

**Comment by jcoupey** [comment]
> I figured out that if i change the time_window:[val,val] to time_windows:[[val:val]]

Good point. The `time_window` key are ignored for the jobs, you should use `time_windows` with an array of time frames.

> This should be solvable

I could not really figure out what you think is wrong since you get a solution with all jobs assigned? Can you expand on what is actually wrong with that solution or what you'd expect instead?

**Comment by devilmastah** [comment]
Hey, i will try to make it visual later today. 
The problem is mainly that i get some big overlaps of orders while there would be more cars available.

**Comment by jcoupey** [comment]
> i am a bit baffled by the drivetime differences that happen sometimes

There is no such thing as "drivetime from Vroom" as all costs are retrieved using matrix requests VROOM sends to the underlying engine (ORS in your case). If you notice differences, then something is wrong somewhere, either in the problem description or the solution interpretation.

It's a bit hard to follow from just a screenshot but my guess is that it is a matter of presentation of the solution. The green tasks seem to happen ASAP. This may be nice from some point of view but increases the waiting time inside the route. The red blocks are tighter which makes sense with the way we present routes in output: end route ASAP, then browse the route back to determine when is the latest possible time to start that matches this end time. This approach minimizes waiting times inside the route.

**Comment by devilmastah** [comment]
Thanks for the insights. 
The second part you describe makes a lot of sense, seems like vroom is not a good match for what we want then :) 
Waiting times are not really an issue for us, basicly we take orders in certain timeslots, a customer wants us there at the beginning of the slot but we allow for 30 minutes of slack. And if there is time usually its extra profit (we cook on location for parties).

Anyhow, thanks for the reactions, i will close this ticket :)
